cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/NewCommand.java

```diff
@@ -105,36 +104,109 @@ public void execute() {
         if (!(1 == argList.size())) {
             CommandUtil.printError(errStream,
                     "too many arguments",
-                    "bal new <project-name>",
+                    "bal new <project-path>",
                     true);
             CommandUtil.exitError(this.exitWhenFinish);
             return;
         }
 
-        // If the current directory is a ballerina project, fail the command.
-        if (ProjectUtils.isBallerinaProject(this.userDir)) {
-            CommandUtil.printError(errStream,
-                    "directory is already a Ballerina project.",
-                    null,
-                    false);
-            CommandUtil.exitError(this.exitWhenFinish);
-            return;
+        Path argPath = Paths.get(argList.get(0));
+        Path packagePath = argPath;
+        Path currentDir = Paths.get(System.getProperty(ProjectConstants.USER_DIR));
+        try {
+            Path relativeToCurrentDir = Paths.get(currentDir.toString(), packagePath.toString()).normalize();
+            if (Files.exists(relativeToCurrentDir) || Files.exists(relativeToCurrentDir.getParent())) {
+                packagePath = relativeToCurrentDir;
+            }
+        } catch (InvalidPathException ignored) {
+            // If the path is not a valid path, use the given path as it is.
         }
 
-        String packageName = argList.get(0);
-        Path path = userDir.resolve(packageName);
-        // Check if the directory or file exists with the given project name
-        if (Files.exists(path)) {
-            CommandUtil.printError(errStream,
-                    "destination '" + path.toString() + "' already exists.",
-                    "bal new <project-name>",
-                    true);
-            CommandUtil.exitError(this.exitWhenFinish);
-            return;
+        CommandUtil.setPrintStream(errStream);
+        Path packageDirectory;
+        String packageName;
+        boolean balFilesExist = false;
+
+        // Check if the given path is a valid path
+        if (Files.exists((packagePath))) {
+            // If the given path is a ballerina project, fail the command.
+            if (ProjectUtils.isBallerinaProject(packagePath)) {
+                CommandUtil.printError(errStream,
+                        "directory is already a Ballerina project.",
+                        null,
+                        false);
+                CommandUtil.exitError(this.exitWhenFinish);
+                return;
+            }
+
+            //Check if package files/directories other than .bal files exist
```

Comment:
It is better to add a fullstop, like in L122
```suggestion
            //Check if package files/directories other than .bal files exist.
```